src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -109,10 +109,7 @@ private void ProcessEndpointParameterSource(Endpoint endpoint, ISymbol symbol, I
             }
             // Complex form binding is only supported in RDF because it uses shared source with Blazor that requires dynamic analysis
             // and codegen. Emit a diagnostic when these are encountered to avoid producing buggy code.
-            else if (!(SymbolEqualityComparer.Default.Equals(Type, wellKnownTypes.Get(WellKnownType.Microsoft_Extensions_Primitives_StringValues))
-                    || Type.SpecialType == SpecialType.System_String
-                    || TryGetParsability(Type, wellKnownTypes, out var _)
-                    || (IsArray && TryGetParsability(ElementType, wellKnownTypes, out var _))))
+            else if (RequiresComplexFormBinding(wellKnownTypes))
```

Comment:
```suggestion
            else if (!HasSimpleBinding(wellKnownTypes))
```

The `!` inside the `RequiresComplexFormBinding` is still hard to parse with the nested parentheses. I think moving it outside the method would make it clearer.

Reply:
Fixed.